src/libstd/io/mod.rs

```diff
@@ -219,6 +318,29 @@ pub trait Read {
     ///
     /// The returned adaptor also implements `Read` and will simply borrow this
     /// current reader.
+    ///
+    /// # Examples
+    ///
+    /// [`File`][file]s implement `Read`:
+    ///
+    /// [file]: ../std/fs/struct.File.html
+    ///
+    /// ```
+    /// use std::io;
+    /// use std::fs::File;
+    /// use std::io::Read;
+    ///
+    /// # fn foo() -> io::Result<()> {
+    /// let mut f = try!(File::open("foo.txt"));
+    /// let mut buffer = String::new();
+    ///
+    /// let reference = f.by_ref();
```

Comment:
This example may want to show off `by_ref` in terms of composition, for example something like:

```
// read at most 5 bytes
f.by_ref().take(5).read_to_end(&mut buf);

// original file still usable, read the rest
f.read_to_end(&mut other_buf);
```